### DISCUSSION TITLE
Windows Event Data section

### DISCUSSION THREAD

**Discussion by frack113** [SigmaHQ/sigma-specification#21] [Ideas]
Ref https://github.com/SigmaHQ/sigma/issues/3622
Some Windows Eventid don't have field name.
`data` field name should be add to [Taxonomy_1_2_0.md](https://github.com/SigmaHQ/sigma-specification/blob/main/Taxonomy_1_2_0.md).

**Comment by phantinuss** [comment]
I am not sure I understand correctly. The rule in question just uses a keyword search and does not specify on which EventID it should match. I don't see an issue with the rule.

What would the `data` field name be and what would be its purpose/what would it map to?

**Reply by nasbench** [reply]
In classic Event logs they don't have an ETW manifest to translate the fields so all the fields of an event are bundled in a field called "Data". We can use this field in SIGMA/Aurora but for tools that use these rules, they either need to collect it as is and call it a field or create parsers to separate the data into separate fields.

Since we do use it. I think it should be part of the SIGMA spec

**Reply by phantinuss** [reply]
I think this is exactly what the keyword search is intended for. It is a search over all fields which can be used if fields are not defined properly. 

Data elements without the name attribute (as seen in Windows application log) is just unstructured data. We cannot know beforehand how a back end would parse these events and I think it is not needed to complicate things by introducing a field that actually is not a real field.

**Reply by phantinuss** [reply]
Maybe there is a solution to the issue but I don't like using `data` as a field if it is, from my understanding, not a real field. Using the keyword search we at least do not cover the fact that the data is not structured well.

**Reply by nasbench** [reply]
The issue with the keyword is you can't use `modifiers` and most of the time when we write rules for this we use a combination of stuff. And "Data" is actually a field in the `Event`. 

Example:
 - b3d57a5c-c92e-4b48-9a79-5f124b7cf964
 - 7f103213-a04e-4d59-8261-213dddf22314

**Reply by phantinuss** [reply]
It is an unnamed field. Every event data is inside a `data` element. So it is still unstructured data and the field can (and does) exist multiple times. I am not sure if that itself is an issue.

**Reply by thomaspatzke** [reply]
> The issue with the keyword is you can't use `modifiers`

Wrong 😉

```
detection:
    selection:
        |contains: value
```

Does the trick 😉

Regarding the main question, I'm a bit confused about the `Data` element. Partially they seem to be named, e.g. [here for Security/4688](https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4688) with `<Data Name="...">` here I think the usage of the field name from the `Name` attribute is already practiced a lot in open source Sigma rules. On the other side the `Data` elements can also be unnamed, like described [here](https://learn.microsoft.com/de-de/archive/blogs/jonathanalmquist/datacontextdataitemeventdatadataitemeventdatadata). I assume this discussion is about the latter case.

From my perspective, it would be consistent to use `Data` for such cases as Sigma field name. This should be documented as special case in the taxonomy documentation and that it must be handled accordingly by rule preprocessingt or conversion backends, E.g. in Splunk this can be simply accomplished by mapping to `EventData_Xml`.

**Reply by phantinuss** [reply]
We need to clarify this special case in the spec then as well:
https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_1_0_1.md#field-usage
In my opinion this should be 3. iii) then.

**Reply by thomaspatzke** [reply]
Argh...we should stop to do versioning with file names (link was broken) 😐

I think this is more appropriate in the modifiers section because this is where the special case applies. Normal keyword detections are expressed as list. Or in both places (despite the fact I dislike redundancy), such that it doesn't gets missed.

**Comment by nasbench** [comment]
This has been resolved and sigma rules in the repo have been modified.
